File: phase00-foundations/max_min_in_array.js
// Problem: Find maximum and minimum in an array

const findMaxMinNums = (nums) => {
  let minimum = +Infinity;
  let maximum = -Infinity;

  for (let num of nums) {
    if (num > maximum) {
      maximum = num;
    }

    if (num < minimum) {
      minimum = num;
    }
  }

  return {
    maximum,
    minimum,
  };
};

const result = findMaxMinNums([2, 7, 11, 15, 6, 8]);
console.log(result);

File: repractice/2026-07-13/max_min_in_array.py
# Find maximum and minimum in an array


def maxMinNumsInArr(numArr):
    maximum = float("-inf")
    minimum = float("inf")

    for num in numArr:
        if num > maximum:
            maximum = num
        if num < minimum:
            minimum = num

    return maximum, minimum


testNumArr = [2, 7, 11, 15, 6, 8]
print(maxMinNumsInArr(testNumArr))
